syntax = "proto3";

package HW1;

service UserService {
    rpc Registrazione_utente (Registrazione_utente_Request) returns (Registrazione_utente_Reply) {}
    rpc Aggiornamento_utente (Aggiornamento_utente_Request) returns (Aggiornamento_utente_Reply) {}    
    rpc Delete_utente (Delete_utente_Request) returns (Delete_utente_Reply) {}
    rpc Get_Last_Value(Get_Last_Value_utente_Request) returns (Get_Last_Value_utente_Reply){}
    rpc Get_Media(Get_Media_utente_Request) returns (Get_Media_utente_Reply){}
}

message Registrazione_utente_Request {
    string email = 1;
    string ticker=2;
}

message Registrazione_utente_Reply {
  string messaggio = 1;
  string stato_registrazione=2;
}

message Aggiornamento_utente_Request {
    string email = 1;
    string ticker=2;
}

message Aggiornamento_utente_Reply{
    string messaggio=1;
    string stato_aggiornamento=2;
}

message Delete_utente_Request{
    string email=1;
}

message Delete_utente_Reply{
    string messaggio=1;
    string stato_delete=2;
}

message Get_Last_Value_utente_Request{
    string email=1;
    string ticker=2;
}

message Get_Last_Value_utente_Reply{
    double res=1;
}

//num_valori= numero N di valori per fare la media
message Get_Media_utente_Request{
    string email=1;
    string ticker=2;
    int32 num_valori=3; 
}

message Get_Media_utente_Reply{
    double media=1;
}

